services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/PermissionService.kt

```diff
@@ -16,43 +23,26 @@ class PermissionService(
             it.hankeId
         }
 
-    fun hasPermission(hankeId: Int, userId: String, permission: PermissionCode): Boolean {
-        val kayttooikeustaso =
-            permissionRepository.findOneByHankeIdAndUserId(hankeId, userId)?.kayttooikeustaso
-        return hasPermission(kayttooikeustaso, permission)
-    }
+    fun hasPermission(hankeId: Int, userId: String, permission: PermissionCode): Boolean =
```

Comment:
Do you think there is a way to make these work with hankeTunnus instead of hankeId? Currently the way the majority of our endpoints work is that we receive hankeTunnus in the controller, then find hankeId and then call this service. This is somewhat unrelated to this pr, but it is something that I've been wondering.

Reply:
An easy solution would be to overload this:
```
fun hasPermission(hankeTunnus: String, userId: String, permission: PermissionCode): Boolean =
    hasPermission(hankeService.getHankeId(hankeTunnus), userId, permission)
```
Or maybe we would need to use hankeRepository instead of the service to avoid a circular dependency.

In a broader view, there's not much point in having both hankeId and hankeTunnus, since hankeTunnus is created for every hanke at first save. We could refactor the database and code to use hankeTunnus as the primary key for the hanke and get rid of hankeId. This wouldn't be the simplest refactoring, but it might be simpler than it seems at first glance.